mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,94 @@
+from azure.storage.blob import BlobClient
+
+import os
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.utils.string_utils import strip_suffix
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+from mlflow.utils.rest_utils import call_endpoint, extract_api_info_for_service
+from mlflow.protos.databricks_artifacts_pb2 import DatabricksMlflowArtifactsService
+from mlflow.protos.databricks_artifacts_pb2 import GetCredentialsForWrite, GetCredentialsForRead
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.protos.service_pb2 import MlflowService, ListArtifacts
+
+_PATH_PREFIX = "/api/2.0"
+
+
+class DatabricksArtifactRepository(ArtifactRepository):
+    """
+    SOMETHING : TYPING TILL IT WORKS LOL
+    """
+
+    def __init__(self, artifact_uri):
+        super(DatabricksArtifactRepository, self).__init__(artifact_uri)
+
+    def _extract_run_id(self, artifact_uri):
+        return artifact_uri.lstrip('/').split('/')[4]
+
+    def _call_endpoint(self, service, api, json_body):
+        _METHOD_TO_INFO = extract_api_info_for_service(service, _PATH_PREFIX)
```

Comment:
This `extract_api_info_for_service` method is a bit expensive (https://github.com/mlflow/mlflow/blob/c4e0cbdf5c92b3b81f7080fff4a974114605d2db/mlflow/utils/rest_utils.py#L115). Can we instead define a module-level `_SERVICE_AND_METHOD_TO_INFO` dictionary as:
```
_SERVICE_AND_METHOD_TO_INFO = {
    service: extract_api_info_for_service(service, _PATH_PREFIX)
    for service in [MlflowService, DatabricksMlflowArtifactsService]
}
```

We can then fetch the info as `_SERVICE_AND_METHOD_TO_INFO[service][api]`, without performing proto extraction operations each time this method is called.

Reply:
Done.